docs/dev/compiler-toolchain/solidity.md

```diff
@@ -0,0 +1,199 @@
+
+#### Free-form output
+
+This output format is utilized in Yul and LLVM IR compilation modes. These modes only support compiling a single file at
+the moment. Only `--asm` and `--bin` output flags are supported, so this mode can be useful for debugging and prototyping.
+
+#### CLI reference
+
+- `--version`  
+  Print the version and exit.  
+
+- `<input_files>`  
+  Specify the input file paths.  
+  Multiple Solidity files can be passed in the default Solidity mode.  
+  Yul and LLVM IR modes currently support only a single file.  
+
+- `--base-path <path>`  
+  Set the given path as the root of the source tree instead of the root of the filesystem.  
+  Passed to `solc` without changes.  
+
+- `--include-path <path>`  
+  Make an additional source directory available to the default import callback. Can be used multiple times. Can only be used if the base path has a non-empty value.  
+  Passed to `solc` without changes.  
+
+- `--allow-paths <path1,path2,...>`  
+  Allow a given path for imports. A list of paths can be supplied by separating them with a comma.  
+  Passed to `solc` without changes.  
+
+- `-o`, `--output-dir <path>`  
+  Create one file per component and contract/file at the specified directory, if given.  
+
+- `--overwrite`  
+  Overwrite existing files (used together with -o).  
+
+- `-O`, `--optimization <level>`  
+  Set the optimization parameter -O[0 | 1 | 2 | 3 | s | z].  
+  Use `3` for best performance and `z` for minimal size. See [LLVM optimizer](./llvm.md#optimizer).  
+
+- `--solc <path>`  
+  Specify the path to the `solc` executable. By default, the one in `${PATH}` is used.  
+  Yul mode: `solc` is used for source code validation, as `zksolc` itself assumes that the input Yul is valid.  
+  LLVM IR mode: `solc` is unused.  
+
+- `-l`, `--libraries <string>`  
+  Specify addresses of deployable libraries. Syntax: `<libraryName>=<address> [, or whitespace] ...`.  
+  Addresses are interpreted as hexadecimal strings prefixed with `0x`.  
+
+- `--combined-json <options>`  
+  Output a single JSON document containing the specified information.  
+  Available arguments: `abi`, `hashes`, `metadata`, `devdoc`, `userdoc`, `storage-layout`, `ast`, `asm`, `bin`, `bin-runtime`.  
+
+- `--standard-json`  
+  Switch to standard JSON input/output mode. Read from stdin, write the result to stdout.  
+  This is the default used by the hardhat plugin.  
+
+- `--yul`  
+  Switch to the Yul mode.  
+  Only one input Yul file is allowed.  
+  Cannot be used with the combined and standard JSON modes.  
+
+- `--llvm-ir`  
+  Switch to the LLVM IR mode.  
+  Only one input LLVM IR file is allowed.  
+  Cannot be used with the combined and standard JSON modes.  
+
+- `--force-evmla`  
+  Forcibly switch to the EVM legacy assembly pipeline.  
+  It is useful for older revisions of `solc` 0.8, where Yul was considered highly experimental and contained more bugs than today.  
```

Comment:
```suggestion
  Useful for older revisions of `solc` 0.8, where Yul was considered highly experimental and contained more bugs than today.  
```

Reply:
revisions?

Reply:
In SemVer it is actually called `patch` (the 3rd number in 0.8.19), but this term is a little ambiguous as it can refer to any change made to any program, like a commit in the official repo or even something kept on someone's local machine and applied privately before usage.

Revision is a little broader term although not entirely correct when describing SemVer, but it more precisely describes the solc entities we're referring to.

Reply:
would version be a better term?

Reply:
I prefer 'revision', because we have 0.8 as a version in this sentence; and revision is 'smaller' than version, that is one version can have many revisions.

Reply:
ok sgtm